src/titiler/core/titiler/core/factory.py

```diff
@@ -812,6 +824,299 @@ def wmts(
                 },
                 media_type=MediaType.xml.value,
             )
+        
+    def wms(self):  # noqa: C901
+        """
+        Registers /wms endpoint.
+        
+        Notes:
+            - Not sure where to place WMS helper function. Is here fine?"""
+        
+        
+        # The following wms params are ignored: styles, dpi, map_resolution, forma_options, transparent
+        base_wms_params = (
+            "service",
+            "version",
+            "request",
+            "bbox",
+            "crs",
+            "srs",
+            "width",
+            "height",
+            "layers",
+            "format",
+        )
+        # Ignoring: ['styles', 'dpi', 'map_resolution', 'format_options', 'transparent']
+
+        @staticmethod
+        def process_wms_params(request: Request):
```

Comment:
maybe this whole function could be a Dependency we inject into the wms endpoint

Reply:
```python
@dataclass
class WMSParams(DefaultDependency):

    request: str
    version: str

    layers: Optional[List[str]]
    bbox: Optional[List[float]]
    crs: Optional[CRS]
    height: Optional[int]
    width: Optional[int]
    format: Optional[str]

    def __init__(
        self,
        request: Request,
    ):
        # WMS params use `,` delimited list so we can consider the query-parameter
        # as a dictionary
        req = {
            k.lower(): v
            k, v in for request.query_params
        }

        # Request is mandatory
		if not request_type := req.get("request"):
            raise HTTPException(status_code=400, detail="Missing WMS REQUEST parameter."

        self.request = request_type.lower()

        ogc_spec_version = req.get("version")

        if self.request == "getcapabilities":
            # Required parameters:
            # - SERVICE=WMS
            # - REQUEST=GetCapabilities
            # Optional parameters: VERSION, FORMAT, UPDATESEQUENCE
            if not service := req.get("service"):
                raise HTTPException(status_code=400, detail="Missing WMS SERVICE parameter.")

            if not service.lower() == "wms":
                raise HTTPException(status_code=400, detail=f"Invalid 'SERVICE' parameter: {service}. Only 'wms' is accepted")

            self.version = ogc_spec_version or "1.3.0"

        elif self.request == "getmap":
            # Required parameters:
            # - VERSION
            # - REQUEST=GetMap,
            # - LAYERS
            # - STYLES
            # - CRS
            # - BBOX
            # - WIDTH
            # - HEIGHT
            # - FORMAT
            # Optional parameters: TRANSPARENT, BGCOLOR, EXCEPTIONS, TIME, ELEVATION, ...

            # List of required parameters (styles is excluded)
            req_keys = {"version", "request", "layers", "crs", "bbox", "width", "height", "format"}

            input_keys = set(req.keys())
            intrs = input_keys.intersection(req_keys)
            missing_keys = req_keys.difference(intrs)

            if len(missing_keys) > 0:
                raise HTTPException(status_code=500, detail=f"Missing 'GetMap' parameters: {missing_keys}")

            self.version = ogc_spec_version

            self.layers = list(req["layers"].split(","))
            if not self.layers:
                raise HTTPException(status_code=400, detail=f"Invalid 'LAYERS' parameter: {req['layers']}."

            self.bbox = list(map(float, req["bbox"].split(",")))
            if len(self.bbox) != 4:
                raise HTTPException(
                    status_code=400,
                    detail="Invalid 'BBOX' parameters: {bbox}. Needs 4 coordinates separated by commas",
                )

            self.crs = CRS.from_user_input(req["crs"])
            self.height = int(req["height"])
            self.width = int(req["width"])
            self.format = WMSMediaType(req["format"]).name

            # TODO: check this
            if self.version == "1.3.0":
                # WMS 1.3.0 is lame and flips the coords of EPSG:4326
                if self.crs == CRS.from_epsg(4326):
                    self.bbox = [
                        self.bbox[1],
                        self.bbox[0],
                        self.bbox[3],
                        self.bbox[2],
                    ]
                # Overwrite CRS:84 with EPSG:4326 when specified
                elif self.crs == CRS.from_user_input("CRS:84"):
                    self.crs = CRS.from_epsg(4326)

        else:
            raise HTTPException(status_code=400, detail=f"Invalid 'REQUEST' parameter: {request_type}. Should be one of ['GetCapabilities', 'GetMap']."

        if self.version not in ["1.0.0", "1.1.0", "1.3.0"]:
            raise HTTPException(
                status_code=400,
                detail=f"Invalid 'VERSION' parameter: {version}. Allowed versions include: '1.0.0', '1.1.0', '1.3.0'",
            )
```

Reply:
These are some pro-level edits